saleor/graphql/product/types/products.py

```diff
@@ -460,7 +460,10 @@ class Meta:
 
     @staticmethod
     def resolve_category(root: models.Product, info):
-        return CategoryByIdLoader(info.context).load(root.category_id)
+        category_id = root.category_id
+        if category_id:
```

Comment:
Can it be blank or only `None`? If that's only `None` I would recommend `is None` as it is a faster comparison